gaia-rs/tests/staking.rs

```diff
@@ -0,0 +1,490 @@
+use crate::utilities::{key_add, run_gaia_and_tendermint, KEY_NAME};
+use bank::cli::tx::{BankCommands, BankTxCli};
+use gaia_rs::{
+    client::{GaiaQueryCommands, GaiaTxCommands, WrappedGaiaQueryCommands, WrappedGaiaTxCommands},
+    query::GaiaQueryResponse,
+    GaiaCoreClient,
+};
+use gears::{
+    commands::client::{
+        keys::KeyringBackend,
+        query::{run_query, QueryCommand},
+        tx::{run_tx, Keyring, LocalInfo, TxCommand},
+    },
+    config::DEFAULT_TENDERMINT_RPC_ADDRESS,
+    tendermint::{rpc::response::tx::broadcast::Response, types::chain_id::ChainId},
+    types::{
+        address::{AccAddress, ValAddress},
+        base::coin::Coin,
+        decimal256::Decimal256,
+        uint::Uint256,
+    },
+};
+use staking::{
+    cli::{
+        query::{
+            DelegationCommand, RedelegationCommand, StakingCommands as QueryStakingCommands,
+            StakingQueryCli, ValidatorCommand,
+        },
+        tx::{StakingCommands, StakingTxCli},
+    },
+    BondStatus, CommissionRatesRaw, CommissionRaw, DelegationResponse, Description,
+    RedelegationResponse, Validator,
+};
+use std::{path::PathBuf, str::FromStr};
+use utilities::ACC_ADDRESS;
+
+#[path = "./utilities.rs"]
+mod utilities;
+
+fn run_tx_local(
```

Comment:
Could we reuse this method in others tests?

Reply:
Do you mean `utilities.rs`? Of course, we can. I suggest  it's  a smaller abstraction that allows to reduce a lot of code. 

I've used it many times in the file.

Reply:
May you use it in this PR? Or make another one